assets/src/application/ui/display/confirm/useConfirmWithButton.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { __ } from '@wordpress/i18n';
+
+import { Button } from '@application/ui/input';
+import { ConfirmPropsWithButton } from './types';
+import useConfirmationDialog from './useConfirmationDialog';
+
+const useConfirmWithButton: React.FC<ConfirmPropsWithButton> = ({ buttonProps, ...props }) => {
```

Comment:
@tn3rb  I've went here with the same approach you've described on Friday with having separate components, like `LinkText`, `LinkIcon`, etc...

Reply:
otherwise would have ended up with a bit messy types and too verbose checks for button props existence